src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Win32.SafeHandles;
+
+using static Interop.Crypt32;
+
+namespace Internal.Cryptography.Pal.Windows
+{
+    internal sealed partial class PkcsPalWindows : PkcsPal
+    {
+        internal PkcsPalWindows()
+        {
+        }
+
+        public sealed override DecryptorPal Decode(byte[] encodedMessage, out int version, out ContentInfo contentInfo, out AlgorithmIdentifier contentEncryptionAlgorithm, out X509Certificate2Collection originatorCerts, out CryptographicAttributeObjectCollection unprotectedAttributes)
+        {
+            return DecryptorPalWindows.Decode(encodedMessage, out version, out contentInfo, out contentEncryptionAlgorithm, out originatorCerts, out unprotectedAttributes);
+        }
+
+        public sealed override string DecodeDocumentName(byte[] encodedDocumentName)
+        {
+            return DecodeOctetString(encodedDocumentName);
```

Comment:
"octet string" in ASN/DER land just means "byte[]"  I know this is the name of the original method in netfx, but maybe the helper method would be better as ReadOctetStringAsUnicode.

I really wonder why they did this as OCTET_STRING when UNICODE_STRING is a text-string type.  But, that's what they did :(